config/webpack/utils/resolvePackage.js

```diff
@@ -0,0 +1,78 @@
+const path = require('path');
+const memoize = require('memoizee');
+const chalk = require('chalk');
+const debug = require('debug')('sku:resolvePackage');
+
+const getProjectDependencies = readFileSync => {
+  let pkg;
+
+  try {
+    pkg = JSON.parse(readFileSync(`${process.cwd()}/package.json`).toString());
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      pkg = {};
+    } else {
+      throw error;
+    }
+  }
+
+  return {
+    ...(pkg.dependencies || {}),
+    ...(pkg.devDependencies || {})
+  };
+};
+
+/**
+ * Resolve a package name to an absolute path.
+ * e.g. my-package -> /Users/me/code/my-project/node_modules/my-package
+ *
+ * Throws if a package is listed in the project's dependencies and cannot be resolved.
+ */
+const createPackageResolver = (fs, resolve) =>
+  memoize(function resolvePackage(packageName) {
+    const cwd = process.cwd();
+
+    // First, look for a local copy of the package. If it exists, we're done.
+    const localPackage = path.join(cwd, 'node_modules', packageName);
+
+    if (fs.existsSync(localPackage)) {
+      debug(`Resolved ${packageName} to ${localPackage}`);
+      return localPackage;
+    }
+
+    // If there's no local package, check to see if the project has it as a dependency. If it does
+    // we'll try and track it down, if not we'll return a naive local path. This prevents webpack
+    // from throwing invalid config errors.
+    const dependencies = getProjectDependencies(fs.readFileSync);
+
+    if (!dependencies[packageName]) {
+      debug(`Resolved ${packageName} to ${localPackage}`);
+      return localPackage;
+    }
+
+    // If we don't find a local copy we use Node's require algorithm to find one.
+    // This branch is used when sku or seek-style-guide have been `npm link`ed into a project,
+    // and also by the seek-style-guide.test.js sku test.
+    // Doing this can produce hard-to-debug build issues, so we log a warning.
+    const resolvedPackage = path.dirname(
+      resolve(`${packageName}/package.json`)
```

Comment:
Will this throw if it doesn't exist? Are we catching this error?

Reply:
Yes we throw. I know the existing behaviour is to always return a path. My rationale here is that if you've listed the package in your dependencies and we can't resolve it, we should throw. Webpack is probably going to throw anyway. It felt weird to return a path when we know it doesn't exist.

Thoughts?

Reply:
I originally ran into this issue because we were trying to resolve `seek-style-guide`, since it's included as a default, regardless of whether a sku consumer has it installed. Would this still happen?

Reply:
I've expanded the comments to make it clearer!